akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/internal/ReplicationImpl.scala

```diff
@@ -103,10 +105,21 @@ private[akka] object ReplicationImpl {
       .withReplicatedEventOriginFilter(new EventOriginFilter(settings.selfReplicaId))
 
     val sharding = ClusterSharding(system)
-    sharding.init(replicatedEntity.entity)
+    val shardRegion = sharding.init(replicatedEntity.entity)
 
     // sharded daemon process for consuming event stream from the other replicas
-    val entityRefFactory: String => EntityRef[Command] = sharding.entityRefFor(replicatedEntity.entity.typeKey, _)
+    // sends events over cluster sharding to the entities
+    val entityRefFactory: String => EntityRef[Command] =
+      if (settings.configureEntityWithoutEnvelope.isEmpty) { entityId: String =>
+        sharding.entityRefFor(replicatedEntity.entity.typeKey, entityId)
+      } else { _: String =>
+        new NoEnvelopeEntityRef[Command](
+          replicatedEntity.entity.typeKey,
+          replicatedEntity.entity.dataCenter,
+          shardRegion.asInstanceOf[ActorRef[Command]] // it's not really ShardEnvelope[Command]
```

Comment:
it's fundamentally tied to ShardingEnvelope, also the `akka.cluster.sharding.typed.ReplicatedEntity`, and that is public together with the ReplicatedEntityProvider